types/lodash/index.d.ts

```diff
@@ -280,9 +280,9 @@ declare namespace _ {
         (value: number): LoDashImplicitWrapper<number>;
         (value: string): LoDashImplicitStringWrapper;
         (value: boolean): LoDashImplicitWrapper<boolean>;
-        (value: number[]): LoDashImplicitNumberArrayWrapper;
-        <T>(value: T[]): LoDashImplicitArrayWrapper<T>;
-        <T extends {}>(value: T): LoDashImplicitObjectWrapper<T>;
+        (value: number[] | null | undefined): LoDashImplicitNumberArrayWrapper;
+        <T>(value: T[] | null | undefined): LoDashImplicitArrayWrapper<T>;
+        <T extends {}>(value: T | null | undefined): LoDashImplicitObjectWrapper<T>;
```

Comment:
Yeah, I thought about this one for a while but couldn't come up with a great solution.

It's true that `_(undefined)` will technically not give you an array wrapper, or object wrapper, or any other wrapper we have defined. `_(undefined).value() === undefined`. However, you WILL get  an array wrapper as soon as you call an array function - it will always be an empty array. `_(undefined).filter({ id: 1 }).value() === []`

I wrote it this way to handle cases like the one below:

```
function getMatchCount(a: string[] | undefined): number {
    return _(a)
        .filter(a => a.indexOf("match") > 5)
        .size();
    // Even if `a` is `undefined`, this function will return 0, not undefined
    // Even _(a).size() would return 0.
}
```

There are 2 main cases (that I can think of) where my definition is not correct:

1. `_(null | undefined).value()   // Output type is number[], but actual value is null or undefined`
2. `let a: string[] | undefined; _(a).value() //Output type is number[], but actual value might be undefined`

In the first case, the output type will be `number[]` because Typescript will choose the first available match (which is `LoDashImplicitNumberArrayWrapper` in this case). But I chose to ignore these 2 cases because I don't see a good reason for anyone to use either of them - why would you create a wrapper around something, then immediately get its value?

I suppose I could at least handle the first case by adding a new definition at the top:

`(value: null | undefined): LoDashImplicitWrapper<null | undefined>;`

But I couldn't come up with a good way to handle case 2. The best idea I had would be to create a new type (e.g. `LoDashImplicitNillableArrayWrapper<T>`) that would be just like `LoDashImplicitArrayWrapper<T>`, except it could return `null | undefined` when you call `value()`. But that would mean nearly doubling the size of our type definitions, because we would need to duplicate all of the function definitions for `LoDashImplicitArrayWrapper<T>` and `LoDashImplicitNillableArrayWrapper<T>`. Same thing for all the other implicit and explicit wrapper types.